src/sqlfluff/rules/L041.py

```diff
@@ -93,9 +101,24 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 edit_segments=edit_segments,
             )
         )
-        # Delete original newlines between select keyword and select clause modifier
-        # and delete the original select clause modifier.
-        fixes.extend((LintFix.delete(s) for s in leading_newline_segments))
+
+        # Delete original newlines and whitespace between select keyword
+        # and select clause modifier.
+
+        # If there is not a newline after the select clause modifier then delete
+        # newlines between the select keyword and select clause modifier.
+        if not trailing_newline_segments:
+            fixes.extend((LintFix.delete(s) for s in leading_newline_segments))
+        # If there is a newline after the select clause modifier then delete both the
+        # newlines and whitespace between the select keyword and select clause modifier.
+        else:
+            fixes.extend(
+                (
+                    LintFix.delete(s)
+                    for s in leading_newline_segments + leading_whitespace_segments
+                )
+            )
```

Comment:
Does this need to be two different statements? Will `leading_whitespace_segments` not just be blank? So any harm in always using the section option?

Reply:
Hey @tunetheweb, I initially tried that and found it broke test 1 & 2. It was incorrectly removing the white space from the trailing `SELECT ELEMENT`.

I’m not at my computer at the moment; but I can provide an example tomorrow!

Reply:
OK let's leave as is then.